lib/resque_bus/adapter.rb

```diff
@@ -11,16 +12,48 @@ def enabled!
         QueueBus::Worker.extend(::QueueBus::Adapters::Resque::RetryHandlers)
       end
 
-      def redis(&block)
+      def redis=(server)
+        @queuebus_redis = nil
+        @queuebus_redis_url = server
+      end
+
+      def queuebus_redis
+        @queuebus_redis ||
+          if @queuebus_redis_url
```

Comment:
It took me a couple of reads to understand what you are doing inside this `if` block. Then I looked at the definition of `Resque.redis=` and it made sense.
I think it will be more readable and simpler to accept only `redis://` url string and do 
```ruby
# Picked from `Resque.redis=`
Redis.connect(url: @queuebus_redis_url, thread_safe: true)
```
or if we want to be as flexible as resque,
to copy the definition of `Resque.redis=` to some method in this class like `_redis_client`.